duckduckgo/iOS
# onboarding experiment 3

Task/Issue URL: https://app.asana.com/0/392891325557410/1117195387371337
Tech Design URL: 
CC:

**Description**:

Tests a new theme selection screen as part of onboarding.

Also adds new "load" pixel to allow for timed pixels, as well as adds a test parameter which is enabled in debug mode so they can be excluded.  Also uses new top level `m` pixel.

**Steps to test this PR**:
Two variants to test:

* `sc`: Summary screen only

1. Click start browsing.  App launches in to home page

* `mr`: Theme selection + Summary screen

1. Continue button disabled until one of the themes selected
1. Skip button moves to Summary screen
1. Selecting a theme has haptic feedback and enabled continue button
1. If a theme was selected (ie test with light) then it is applied after summary screen

Pixels:

*    m_o - on boarding shown
*    m_o_t - fired when themes screen is dismissed regardless of user choices, includes timing parameter
*    m_o_t_d - fired if dark theme selected
*    m_o_t_l - fired if light theme selected
*    m_o_t_s - fired if themes are skipped
*    m_o_s - fired when summary screen is dismissed, includes timing parameter

Devices:

* SE

1. All items should be visible on both screens but spacings adjusted to fit

* Other phones

1. All items shown, spacing better used

* iPad

1. Should behave like a larger phone but centred on screen with outer-padding.  The animation when moving from one screen to another is deliberately clipped.

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
